Source/WebCore/mathml/MathMLFractionElement.cpp

```diff
@@ -42,7 +42,7 @@ WTF_MAKE_ISO_ALLOCATED_IMPL(MathMLFractionElement);
 using namespace MathMLNames;
 
 inline MathMLFractionElement::MathMLFractionElement(const QualifiedName& tagName, Document& document)
-    : MathMLPresentationElement(tagName, document)
+    : MathMLRowElement(tagName, document)
```

Comment:
I wonder whether we could avoid that, but anyway MathMLRowElement::childrenChanged overrides the following methods:

```
    void childrenChanged(const ChildChange&) override;
    bool acceptsMathVariantAttribute() override;
    RenderPtr<RenderElement> createElementRenderer(RenderStyle&&, const RenderTreePosition&) override;
```

* MathMLRowElement::createElementRenderer is overridden in MathMLFractionElement/MathMLScriptsElement and don't call MathMLPresentationElement's implementation, so there is no behavior change.

* MathMLRowElement::acceptsMathVariantAttribute returns true if the element is an mstyle tag. It's ok because it will be false for MathMLFractionElement/MathMLScriptsElement and hasTagName comparison should be fast. But maybe better to override it explicitly to return false.

* MathMLRowElement::childrenChanged do some invalidation for mo children, when the child list changed and is not overridden in MathMLFractionElement/MathMLScriptsElement. I believe that would not change the behavior so that's probably fine as is, but maybe just add a comment in MathMLRowElement::childrenChanged that it would be good to avoid this invalidation for MathMLFractionElement/MathMLScriptsElement if renderer() is non-null and returns "is valid".

Reply:
Thank you for the comments! I've overridden MathMLRowElement::acceptsMathVariantAttribute explicitly in MathMLFractionElement and MathMLScriptsElement. For the 3rd bullet point, I'm not sure what renderer() refers to, please could you elaborate?

Reply:
I see, thanks a lot! Added the comment.